go/arrow/flight/flightsql/client.go

```diff
@@ -218,6 +218,48 @@ func (c *Client) ExecuteSubstraitUpdate(ctx context.Context, plan SubstraitPlan,
 	return updateResult.GetRecordCount(), nil
 }
 
+// ExecuteIngest is for executing a bulk ingestion and only returns the number of affected rows.
+func (c *Client) ExecuteIngest(ctx context.Context, rdr array.RecordReader, reqOptions *ExecuteIngestOpts, opts ...grpc.CallOption) (n int64, err error) {
+	var (
+		desc         *flight.FlightDescriptor
+		stream       pb.FlightService_DoPutClient
+		wr           *flight.Writer
+		res          *pb.PutResult
+		updateResult pb.DoPutUpdateResult
+	)
+
+	cmd := (*pb.CommandStatementIngest)(reqOptions)
+	if desc, err = descForCommand(cmd); err != nil {
+		return
+	}
+
+	if stream, err = c.Client.DoPut(ctx, opts...); err != nil {
+		return
+	}
+
+	wr = flight.NewRecordWriter(stream, ipc.WithAllocator(c.Alloc), ipc.WithSchema(rdr.Schema()))
+	wr.SetFlightDescriptor(desc)
+
+	for rdr.Next() {
+		rec := rdr.Record()
+		wr.Write(rec)
+	}
+
+	if err = stream.CloseSend(); err != nil {
+		return
+	}
+
+	if res, err = stream.Recv(); err != nil {
+		return
+	}
+
+	if err = proto.Unmarshal(res.GetAppMetadata(), &updateResult); err != nil {
+		return
+	}
+
```

Comment:
I believe you have to drain the read-side of the stream here, even if technically the server should only have sent one message

Reply:
Updated. Let me know what you think of the approach. We don't currently ensure the stream is drained in the other DoPut methods FWIW, but I can make changes there as well if it's preferred.